ui/src/app/subnets/views/SubnetsList/SubnetsControls/SubnetsControls.test.tsx

```diff
@@ -35,11 +27,19 @@ it("displays additional information for group by on press", () => {
   expect(
     screen.queryByTestId("subnets-groupby-help-text")
   ).not.toBeInTheDocument();
-  userEvent.click(screen.getByRole("button", { name: "more about group by" }));
+  await userEvent.click(
+    screen.getByRole("button", { name: "more about group by" })
+  );
   expect(screen.getByTestId("subnets-groupby-help-text")).toBeInTheDocument();
 });
 
 it("calls handleSearch with a correct value on user input", async () => {
+  // As of v14 userEvent always returns a Promise and by default it waits for a
+  // setTimeout delay during its execution. Since jest.useFakeTimers() replaces
+  // the original timer functions, userEvent waits indefinitely. We overwrite
+  // this default delay behaviour by setting it to null during setup.
+  jest.useFakeTimers();
+  const user = userEvent.setup({ delay: null });
```

Comment:
@huwshimi @petermakowski just a "gotcha" type thing with the update that might be good to be aware of.

Reply:
Wouldn't it be better to pass advanceTimers function to options (I think that's what they mean in the docs https://testing-library.com/docs/user-event/options/#advancetimers)
```
const user = userEvent.setup({
    advanceTimers: () => jest.advanceTimersByTime(DEFAULT_DEBOUNCE_INTERVAL),
  });
```
Then the manual advance timers can be removed from the test itself and we can await for the assertion instead:

```
  await waitFor(() => expect(handleSearch).toHaveBeenCalledTimes(1));
```

Reply:
That definitely *sounds* like a more robust approach, but the test fails with:
```
    expect(jest.fn()).toHaveBeenCalledTimes(expected)

    Expected number of calls: 1
    Received number of calls: 4
```

Reply:
You're right. It seems that just passing jest.advanceTimersByTime is enough. 
https://github.com/testing-library/user-event/issues/565#issuecomment-1115039532This passes:

https://github.com/canonical-web-and-design/maas-ui/compare/renovate/testing-library-user-event-14.x...petermakowski:userevent-advance-timers

Reply:
But I totally don't mind if we fix this in a separate PR.